extendr-api/src/wrapper/primitive.rs

```diff
@@ -0,0 +1,36 @@
+use super::*;
+
+/// Wrapper for creating and reading Primitive functions.
+///
+/// ```
+/// use extendr_api::prelude::*;
+/// test! {
+///     let robj = r!(Primitive("+"));
+///     assert!(robj.is_primitive());
+///     assert!(!r!(Primitive("not_a_primitive")).is_primitive());
+/// }
+/// ```
+#[derive(Debug, PartialEq, Clone)]
+pub struct Primitive<'a>(pub &'a str);
+
+impl<'a> From<Primitive<'a>> for Robj {
+    /// Make a primitive object, or NULL if not available.
+    /// ```
+    /// use extendr_api::prelude::*;
+    /// test! {
+    ///     let builtin = r!(Primitive("+"));
+    ///     let special = r!(Primitive("if"));
```

Comment:
Should there be a test here? E.g. that `builtin` and `special` are not `NULL`? And maybe add a case where the result is `NULL`?